api/repo/repair.go

```diff
@@ -116,6 +116,35 @@ func RepairRepo(c *gin.Context) {
 		}
 	}
 
+	// get repo information from the source
+	sourceRepo, err := scm.FromContext(c).GetRepo(ctx, u, r)
+	if err != nil {
+		retErr := fmt.Errorf("unable to retrieve repo info for %s from source: %w", sourceRepo.GetFullName(), err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+	// if repo has a name change, then update DB with new name
+	// if repo has a org change, update org as well
+	if sourceRepo.GetName() != r.GetName() || sourceRepo.GetOrg() != r.GetOrg() {
```

Comment:
Discussed in a side conversation but noting it here: we'll want to handle situations like this similarly to how we [currently handle](https://github.com/go-vela/server/blob/3d67002de8f87154fd52e79e562a90f1aacb6894/api/webhook/post.go#L814) `renamed` / `transferred` repository events, specifically adjusting the repository's resources (builds and secrets) to accommodate the name change.

Reply:
good call - is this being implemented as part of this PR or a follow-up PR?

Reply:
good call - is this being implemented as part of this PR or a follow-up PR?